src/device-api/common.js

```diff
@@ -30,7 +30,10 @@ export async function doRestart(appId, force) {
 
 			return deviceState.pausingApply(async () => {
 				return Bluebird.each(app.services, async (service) => {
-					await serviceManager.kill(service, { wait: true });
+					await serviceManager.kill(service, {
```

Comment:
I'm not sure about this, I don't think the API should be making calls to `serviceManager` in the first place. I think we need to figure out a way to do this through providing target state

Reply:
I don't think that's within the scope of this PR tbh. We can resolve the issue with this simple addition.

Reply:
Maybe, but a one line change also validates this code and adds to technical debt, so we need to at least to make a priority issue about refactoring the API code

Reply:
@pipex I believe this was done before using intermediateState but there was an image removal bug, see: https://github.com/balena-os/balena-supervisor/pull/1524

Reply:
I fixed this image removal bug for purge in #1635, we could apply the change here as well.

Reply:
I'll implement the change to use intermediateState instead of serviceManager in a new PR, closing this one as it's no longer relevant though.